### ISSUE TITLE
Links cannot be selected

### ISSUE THREAD

**Issue by guplem** [flyerhq/flutter_chat_ui#225]
# General

## What bug do you experience? 🐞
While trying to select the text of a message that is a link, the "select all" pop-up appears, but nothing is selected if I click it. Neither can I drag the "delimiters of the selection", they disappear.
Text messages without links do not suffer from this issue.

## How can it be reproduced? 🤔
A few steps to define where does the bug occur.
Step 1. Create a basic chat following the guide
Step 2. Try to select the text of a text message containing a link
Step 3. The "selection delimiters" will disappear almost instantly and the "select all" pop-up will appear: 
![image](https://user-images.githubusercontent.com/11029629/155714057-0bf22d3b-2d6e-4cbf-99ea-9826d0f1bd5f.png)
Step 4. Click select all --> Nothing will be selected

## What behavior is expected? 💡
The user should be able to manually determine what it is wanted to be selected and, if "select all" is pressed, all the message should be selected, like what happens with the messages without links: 
![image](https://user-images.githubusercontent.com/11029629/155714564-624bc415-dadf-4363-85b7-1ade65da5cdc.png)

<br />

# Extras
Some other things are causing trouble too, and I do not know if it is related:
 - Links cannot be opened tapping on them. But I do not know if that is the intended behaviour. (Using onMessageTap doesn't let me know which link has been tapped, if there are more than one).
 - handlePreviewDataFetched doesn't seem to work (maybe I do not understand what data fetches from links, but I thought it should show a preview and it isnt)

## Code snippets 📝
This is the main code I am running

```dart
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:flutter/services.dart' show Clipboard, ClipboardData, rootBundle;
import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
import 'package:flutter_chat_ui/flutter_chat_ui.dart';
import 'package:get/get.dart';
import 'package:kedo/plan/plans_data.dart';
import 'package:kedo/plan/plan.dart';
import 'package:kedo/plan/plan_detail_page.dart';
import 'package:kedo/profile/logged_profile_controller.dart';
import 'package:kedo/chat/message/messages_data.dart';
import 'package:kedo/chat/message/message.dart' as kedo;
import 'package:kedo/profile/profile.dart';
import 'package:kedo/profile/profiles_data.dart';
import 'package:parse_server_sdk_flutter/parse_server_sdk.dart';

import 'package:uuid/uuid.dart';

import 'package:kedo/poy/utils.dart';

import '../profile/profile_page.dart';

class ChatPage extends StatefulWidget {
  late Subscription<ParseObject> subscription;

  ChatPage({Key? key, required this.planId}) : super(key: key);

  final String planId;
  final LiveQuery liveQuery = LiveQuery();

  @override
  _ChatPageState createState() => _ChatPageState();
}

class _ChatPageState extends State<ChatPage> {
  Set<types.Message> _messages = <types.Message>{};
  final _user = ProfilesData.getMessageUserAuthorFromProfile(Get.find<LoggedProfileController>().loggedProfile!);

  @override
  void initState() {
    super.initState();
    _loadCachedMessages();
    _downloadMessages();
    _startLiveQuery();
  }

  void _startLiveQuery() async {
    Debug.log("Live Query Started");
    QueryBuilder<ParseObject> query = QueryBuilder<ParseObject>(ParseObject('Message'));
    query.whereEqualTo("plan", (ParseObject('Plan')..set("objectId", widget.planId)).toPointer());
    widget.subscription = await widget.liveQuery.client.subscribe(query);
    widget.subscription.on(LiveQueryEvent.create, (value) async {
      Debug.log("Detected new stored message on the database for the plan ${widget.planId}");
      ParseObject val = value;
      Profile author = (await ProfilesData.getProfile(profileId: val.get<ParseObject>("sender")!.get<String>("objectId")!))!;
      kedo.Message msg = kedo.Message.fromParseObject(val, author);
      MessagesData.registerOnCache(msg);
      setState(() {
        _messages.add(msg.txtMessage);
      });
    });
  }

  @override
  void dispose() {
    widget.liveQuery.client.unSubscribe(widget.subscription);
    super.dispose();
  }

  void _addMessage(types.Message message) {
    MessagesData.addMessage(
      kedo.Message(txtMessage: message as types.TextMessage, planId: widget.planId),
    );
  }

  void _handlePreviewDataFetched(types.TextMessage message, types.PreviewData previewData) {
    final index = _messages.toList().indexWhere((element) => element.id == message.id);
    final updatedMessage = _messages.toList()[index].copyWith(previewData: previewData);

    WidgetsBinding.instance?.addPostFrameCallback((_) {
      setState(() {
        _messages.toList()[index] = updatedMessage;
      });
    });
  }

  void _onMessageTap(context, message) {
    if (message is types.TextMessage) {
      Clipboard.setData(ClipboardData(text: message.text));
      const SnackBar(content: Text('Se ha copiado el mensaje al portapapeles'));
    } else {
      SnackBar(content: Text('Tipo: ${message.type}'));
    }
  }

  void _handleSendPressed(types.PartialText message) {
    final textMessage = types.TextMessage(
      author: _user,
      createdAt: DateTime.now().millisecondsSinceEpoch,
      id: const Uuid().v4(),
      text: message.text,
    );

    _addMessage(textMessage);
  }

  void _downloadMessages() async {
    //Example:
    // final response = await rootBundle.loadString('assets/messages.json');
    // final messages = (jsonDecode(response) as List)
    //     .map((e) => types.Message.fromJson(e as Map<String, dynamic>))
    //     .toList();

    final Set<kedo.Message> downloadedMessages = await MessagesData.getMessagesFromPlan(planId: widget.planId);

    setState(() {
      _messages.addAll(downloadedMessages.map((e) => e.txtMessage).toSet());
    });
  }

  void _loadCachedMessages() {
    final Set<kedo.Message> downloadedMessages = MessagesData.getCachedMessagesFromPlan(planId: widget.planId);

    setState(() {
      _messages.addAll(downloadedMessages.map((e) => e.txtMessage).toSet());
    });
  }

  @override
  Widget build(BuildContext context) {
    return FutureBuilder<Plan?>(
        future: PlansData.getPlan(widget.planId),
        builder: (context, snapshotPlan) {
          if (snapshotPlan.data == null) {
            return const SizedBox.shrink();
          }
          return Scaffold(
            appBar: AppBar(
              title: Text(snapshotPlan.data!.name),
              actions: [
                Padding(
                  padding: const EdgeInsets.only(right: 15.0),
                  child: PopupMenuButton(
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
                    child: const Icon(Icons.more_horiz),
                    onSelected: (result) {
                      switch (result) {
                        case 0:
                          FlutterShortcuts.pushPage(context, PlanDetailPage(plan: snapshotPlan.data!));
                          break;
                      }
                    },
                    itemBuilder: (context) {
                      return [
                        if (snapshotPlan.data != null)
                          const PopupMenuItem(
                            child: Text('Ver el plan'),
                            value: 0,
                          ),
                      ];
                    },
                  ),
                ),
              ],
            ),
            body: SafeArea(
              bottom: false,
              child: Chat(
                showUserAvatars: true,
                messages: _messages.toList().reversed.toList(),
                onPreviewDataFetched: _handlePreviewDataFetched,
                onSendPressed: _handleSendPressed,
                onAvatarTap: (user) => FlutterShortcuts.pushPage(context, ProfilePage(profileId: user.id)),
                usePreviewData: true,
                hideBackgroundOnEmojiMessages: true,
                scrollPhysics: const BouncingScrollPhysics(),
                onMessageTap: _onMessageTap,
                showUserNames: true,
                l10n: const ChatL10nEs(),
                user: _user,
              ),
            ),
          );
        });
  }
}

```

<br />

# Environment info

Please specify the flutter, flutter-chat-ui versions.

flutter: 2.10.2
flutter-chat-ui: ^1.5.5

`flutter doctor -v` output 👇

```sh
[√] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.19043.1526], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components
[√] Android Studio (version 2020.3)
[√] Android Studio (version 2021.1)
[√] Connected device (4 available)
[√] HTTP Host Availability

! Doctor found issues in 1 category.
```

<br />

# Platform
Android emulator and Android 11 (OxygenOS 11.3)

**Comment by felixgabler** [comment]
This should be fixed after #206 was merged. @demchenkoalex I think you can close it.

**Comment by demchenkoalex** [comment]
Thank you! Will be released today in `1.5.6`
